### PR TITLE
Kafka connector for reactive messaging

### LINKED ISSUE #4086: RM: Create new FAT bucket for reactive messaging
This issue should be used for the initial creation of the new FAT bucket. Individual tests should be added as part of the specific issues.

### LINKED ISSUE #7740: Provide Kafka Server FAT Test Mechanism
This issue should look to possibly using TestContainers or Andy Guiberts mp-jee-testing work

### LINKED ISSUE #7782: MicroProfile Reactive Messaging Kafka IncomingConnectorFactory
This issue tracks the final delivery of a Kafka OutgoingIncommingFactory that can be shipped.

### LINKED ISSUE #7783: MicroProfile Reactive Messaging Kafka OutgoingConnectorFactory
This issue tracks the final delivery of a Kafka OutgoingConnectorFactory that can be shipped.

### LINKED ISSUE #7828: Packaging prototype 6 (AN)  - Provide a shim implementation in a jar/classpath
This issue is a recording of an approach discussed with Alasdair
(as best as I can - I may have miscaught some of his vision - in
which case he can correct me.)

Here are all the relevant snippets from the conversation with Alasdair:
```
"pick up the kafka library from the application"
"app packaging is a better approach for your use case"

"a jar that makes static calls to Kafka, and instead of it being bundle loaded we create a classloader that has the app classloader as a parent and side peeks into the message streams runtime so it can see both and this integration jar."

"the custom classloader means if this special integration jar we create needs to see stuff internal to Liberty we don’t have to expose that into the application classspace."

"which is really important"

"dump an interface in your jar file. Have your code call kafka via that interface. Provide a direct, non reflection implementation but put it in a jar file."
```

The shim interface and implementation will be done under this issue. The classloader work will be done under issue #7843

### PR DESCRIPTION
**Pull request by Azquelt**
Add a kafka connector for microprofile reactive messaging.

Fixes #7783 
Fixes #7782 
Fixes #7828 
Fixes #7740 
Fixes #4086

### CODE REVIEW

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Looks like a change is needed to the DelegatingConfig to work with Config-1.4

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    private Message<V> wrapInMessage(ConsumerRecord<K, V> record) {
+        try {
+            return Message.of(record.value(), this.ackTracker.trackRecord(record));
+        } catch (Throwable t) {
+            Tr.error(tc, "Unexpected error: {}", t);
+            throw t;
+        }
+    }
+
+    public void shutdown() {
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java]: This shutdown method is never called. @Azquelt to open a new issue to consider general shutdown/cleanup.

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/kafka/permissions.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+<!--        <actions>*</actions> -->
+<!--      </permission> -->
+     
+<!--      <permission> -->
+<!--        <class-name>java.util.PropertyPermission</class-name> -->
+<!--        <name>*</name> -->
+<!--        <actions>read</actions> -->
+<!--      </permission> -->
+     
+    <permission>
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/kafka/permissions.xml]: We should remove the AllPermission entry

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java (hunk 1 above)
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaInput.java]: #8006

File: dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/kafka/permissions.xml (hunk 2 above)
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.reactive.messaging_fat/fat/src/com/ibm/ws/microprofile/reactive/messaging/fat/apps/kafka/permissions.xml]: Done

**Comment by Azquelt** [comment]
Squashed commits into appropriate sections.

#build

**Comment by Azquelt** [comment]
#build

**Comment by Azquelt** [comment]
Given that the only failures in the last build were in the com.ibm.ws.microproifile.config.1.4_fat bucket and the only changes since then were to that same bucket, here's a build which only runs that bucket: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_Q1exUJcnEemmUaN-HmhpYw

**Pull request merged**